### ISSUE TITLE
`+breaking` handling

### ISSUE THREAD

**Issue by gavin-norman-sociomantic** [sociomantic-tsunami/neptune#129]
When a patch or minor release contains breaking changes, it should be tagged with `+breaking` and an extra explanation included in the release notes.

It should be possible to detect this state based on the release notes folder.

**Comment by leandro-lucarella-sociomantic** [comment]
Mmmm, so any patch release that has `relnotes` should be considered breaking? I think that might be a bit extreme... I think from time it might be useful to add a release note even if there are no breaking changes. Maybe add a new type of change, like `relnotes/name.breaking.md`?

**Comment by nemanja-boric-sociomantic** [comment]
Perhaps we should just check for `relnotes/name.migration.md` - what we normally use for breaking changes.

**Comment by leandro-lucarella-sociomantic** [comment]
Wait, actually `name.migration.md` should do, that indicates breaking changes anyway...

**Comment by leandro-lucarella-sociomantic** [comment]
Still there seems to be a place lacking for general notes, if we were to add any.

**Comment by gavin-norman-sociomantic** [comment]
Yeah, I was imagining checking for `relnotes/*.migration.md`.

>Still there seems to be a place lacking for general notes, if we were to add any.

You mean like a higher level explanation of _why_ there are breaking changes in a patch/minor release?

**Comment by gavin-norman-sociomantic** [comment]
We could introduce a general `relnotes/intro.md` file (or something) that is always pasted at the top of the release notes, if present.

**Comment by nemanja-boric-sociomantic** [comment]
Better make it `relnotes/*.intro.md`, to avoid conflicts hell, if more than one PR has to do it.

**Comment by mihails-strasuns-sociomantic** [comment]
> You mean like a higher level explanation of why there are breaking changes in a patch/minor release?

I feel like that belongs to specific migration notes, as you may have several each with own rationale.

**Comment by gavin-norman-sociomantic** [comment]
Well, I think the rationale for making breaking changes in a non-major release is kind of a different (and higher-level) matter to the description of the migration procedure for each breaking change.

**Comment by mihails-strasuns-sociomantic** [comment]
Rationale for having breaking change per se is always the same, so it could be part of pre-formatted message.

**Comment by gavin-norman-sociomantic** [comment]
Right, it could be inserted automatically by `neptune-release` upon noticing this situation.

**Comment by gavin-norman-sociomantic** [comment]
Actually, right, it's akin to the standard "this is a patch release, that means etc".

**Comment by mihails-strasuns-sociomantic** [comment]
Yes, exactly.

**Comment by leandro-lucarella-sociomantic** [comment]
Yes, I was referring more to some sort of intro, but maybe it would be better to do that at the time of release, because I don't think it makes sense to do in advance. I could see like highlighting a couple of important changes or saying what is the release focus on.

Not a big deal though, and truly out of the scope of this issue :)

**Comment by leandro-lucarella-sociomantic** [comment]
So in summary, yeah I think we agree any patch or minor release with `relnotes/*.migration.md` notes should be marked automatically as a `+breaking`.
